app/jobs/job_archiver_job.rb

```diff
@@ -8,13 +8,8 @@ def perform(*args)
   def archive_jobs
     @jobs = Job.all
     @jobs.each do |j|
-      if DateTime.now > j.exp_date
-        j.archived = true
-        j.save
-      else
-        j.archived = false
-        j.save
-      end
+      if (DateTime.now > j.exp_date) then j.archived=true else j.archived=false end
```

Comment:
This isn't a one line this is just a block that has been fitted on one line. 
Take a look at. [Proper usage of Ruby statement modifiers](http://stackoverflow.com/a/1962881/2396304
)

As for making it read more like english modify the job model so that the current block then becomes.
```ruby
@jobs.each do |job|
  job.archive! if job.expired?
end
```